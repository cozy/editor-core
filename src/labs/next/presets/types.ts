import { FeatureFlags } from '../../../plugins/feature-flags-context/types';

export interface EditorPresetProps {
  excludes?: Set<string>;
  experimental?: Array<string>;
  featureFlags?: FeatureFlags;
}
