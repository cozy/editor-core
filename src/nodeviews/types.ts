import type { Node as PMNode } from '@atlaskit/editor-prosemirror/model';

export interface ReactNodeProps {
  selected: boolean;
}

export type ProsemirrorGetPosHandler = () => number | undefined;
export type getPosHandler = getPosHandlerNode | boolean;
export type getPosHandlerNode = () => number | undefined;
export type ReactComponentProps = { [key: string]: unknown };
export type ForwardRef = (node: HTMLElement | null) => void;
export type shouldUpdate = (nextNode: PMNode) => boolean;
