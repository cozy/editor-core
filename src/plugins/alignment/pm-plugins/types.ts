export type AlignmentState = 'start' | 'end' | 'center';
export type AlignmentPluginState = {
  align: AlignmentState;
  isEnabled?: boolean;
};
