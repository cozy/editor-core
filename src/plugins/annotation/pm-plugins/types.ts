import type { DecorationSet } from '@atlaskit/editor-prosemirror/view';
import type {
  EditorState,
  SelectionBookmark,
} from '@atlaskit/editor-prosemirror/state';
import type { PortalProviderAPI } from '@atlaskit/editor-common/portal-provider';
import type { EditorAnalyticsAPI } from '@atlaskit/editor-common/analytics';
import type {
  Dispatch,
  EventDispatcher,
} from '@atlaskit/editor-common/event-dispatcher';
import type { InlineCommentAnnotationProvider, AnnotationInfo } from '../types';

export enum ACTIONS {
  UPDATE_INLINE_COMMENT_STATE,
  SET_INLINE_COMMENT_DRAFT_STATE,
  INLINE_COMMENT_UPDATE_MOUSE_STATE,
  INLINE_COMMENT_CLEAR_DIRTY_MARK,
  ADD_INLINE_COMMENT,
  INLINE_COMMENT_SET_VISIBLE,
  CLOSE_COMPONENT,
  SET_SELECTED_ANNOTATION,
}

export interface InlineCommentPluginOptions {
  dispatch: Dispatch;
  eventDispatcher: EventDispatcher;
  portalProviderAPI: PortalProviderAPI;
  provider: InlineCommentAnnotationProvider;
  editorAnalyticsAPI: EditorAnalyticsAPI | undefined;
}
export interface InlineCommentMouseData {
  isSelecting: boolean;
}

export type InlineCommentMap = { [key: string]: boolean };
export type InlineCommentAction =
  | {
      type: ACTIONS.UPDATE_INLINE_COMMENT_STATE;
      data: InlineCommentMap;
    }
  | {
      type: ACTIONS.SET_INLINE_COMMENT_DRAFT_STATE;
      data: {
        drafting: boolean;
        editorState: EditorState;
      };
    }
  | {
      type: ACTIONS.INLINE_COMMENT_UPDATE_MOUSE_STATE;
      data: {
        mouseData: InlineCommentMouseData;
      };
    }
  | { type: ACTIONS.INLINE_COMMENT_CLEAR_DIRTY_MARK }
  | { type: ACTIONS.CLOSE_COMPONENT }
  | {
      type: ACTIONS.ADD_INLINE_COMMENT;
      data: {
        drafting: boolean;
        inlineComments: InlineCommentMap;
        editorState: EditorState;
        selectedAnnotations: AnnotationInfo[];
      };
    }
  | { type: ACTIONS.INLINE_COMMENT_SET_VISIBLE; data: { isVisible: boolean } }
  | {
      type: ACTIONS.SET_SELECTED_ANNOTATION;
      data: { selectedAnnotations: AnnotationInfo[] };
    };

export type InlineCommentPluginState = {
  annotations: InlineCommentMap;
  selectedAnnotations: AnnotationInfo[];
  dirtyAnnotations?: boolean;
  mouseData: InlineCommentMouseData;
  draftDecorationSet?: DecorationSet;
  bookmark?: SelectionBookmark;

  // Denotes if annotations are allowed to be create on empty nodes or nodes of whitespace (Confluence spec)
  disallowOnWhitespace: boolean;

  // Allow users to hide inline comments during editing
  isVisible: boolean;
  skipSelectionHandling: boolean;
};
