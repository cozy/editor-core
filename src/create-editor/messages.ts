import { defineMessages } from 'react-intl-next';
export const editorMessages = defineMessages({
  editorAssistiveLabel: {
    id: 'fabric.editor.editorAssistiveLabel',
    defaultMessage: 'Main content area, start typing to enter text.',
    description:
      'Text that is read out by screen reader when the main editor is in focus',
  },
});
