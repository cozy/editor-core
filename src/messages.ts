// Common Translations will live here
import commonMessages, {
  linkMessages,
  linkToolbarMessages,
  statusMessages,
  dateMessages,
} from '@atlaskit/editor-common/messages';

export { linkMessages };
export { statusMessages };
export { dateMessages };
export { linkToolbarMessages };

export default commonMessages;
