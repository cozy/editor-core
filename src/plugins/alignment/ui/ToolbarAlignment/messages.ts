import { defineMessages } from 'react-intl';

export const messages = defineMessages({
  alignment: {
    id: 'fabric.editor.alignment',
    defaultMessage: 'Alignment',
    description: 'Aligns text',
  },
});
