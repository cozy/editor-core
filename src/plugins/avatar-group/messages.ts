import { defineMessages } from 'react-intl-next';

export const avatarGroupMessages = defineMessages({
  editors: {
    id: 'fabric.editor.editors',
    defaultMessage: 'Editors',
    description:
      'classifying the people that are currently editing the document',
  },
});
