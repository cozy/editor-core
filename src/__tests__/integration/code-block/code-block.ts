import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { getDocFromElement, editable } from '../_helpers';
import { EditorAppearance } from '../../../types';
import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';
import { messages } from '../../../plugins/block-type/messages';
import { codeBlockSelectors } from '../../__helpers/page-objects/_code-block';

const floatingToolbarLanguageSelector = 'div[aria-label="Floating Toolbar"]';

['comment', 'full-page'].forEach(editor => {
  // https://product-fabric.atlassian.net/browse/ED-5564
  // Fix wrong ADF for code block when language is selected
  BrowserTestCase(
    `code-block: produces correct ADF after language change for ${editor}`,
    { skip: ['safari', 'edge'] },
    async (client: any, testName: string) => {
      const page = await goToEditorTestingExample(client);

      await mountEditor(page, {
        appearance: editor as EditorAppearance,
      });

      await page.click(`[aria-label="${messages.codeblock.defaultMessage}"]`);
      await page.waitForSelector(codeBlockSelectors.languageSelectInput);
      await page.type(codeBlockSelectors.languageSelectInput, ['javascript']);
      await page.keys('Return');

      const doc = await page.$eval(editable, getDocFromElement);
      expect(doc).toMatchCustomDocSnapshot(testName);
    },
  );

  BrowserTestCase(
    `code-block: code block language is preserved after floating toolbar loses and gains focus for ${editor}`,
    { skip: ['safari', 'edge'] },
    async (client: any) => {
      const page = await goToEditorTestingExample(client);

      await mountEditor(page, {
        appearance: editor as EditorAppearance,
      });

      // Insert code block
      await page.click(`[aria-label="${messages.codeblock.defaultMessage}"]`);
      await page.waitForSelector(codeBlockSelectors.languageSelectInput);
      // Change code block language
      await page.type(codeBlockSelectors.languageSelectInput, ['javascript']);
      await page.keys('Return');
      await page.click(editable);
      // Unfocus code block (so floating toolbar hides)
      await page.keys(['ArrowRight', 'ArrowRight']);
      await page.type(editable, 'test paragraph');
      // Focus code block again
      await page.click('pre');
      // Check that the language is still selected
      await page.waitForSelector(floatingToolbarLanguageSelector);
      const language = await page.getText(floatingToolbarLanguageSelector);
      expect(language.trim()).toEqual('JavaScript');
    },
  );

  BrowserTestCase(
    `code-block: code block selected language correctly changes when moving selection directly from one code block to another for ${editor}`,
    { skip: ['safari', 'edge'] },
    async (client: any) => {
      const page = await goToEditorTestingExample(client);

      await mountEditor(page, {
        appearance: editor as EditorAppearance,
      });

      // Insert code block
      await page.click(`[aria-label="${messages.codeblock.defaultMessage}"]`);
      await page.waitForSelector(codeBlockSelectors.languageSelectInput);

      // Change code block language
      await page.type(codeBlockSelectors.languageSelectInput, ['javascript']);
      await page.keys('Return');
      await page.click(editable);
      // Move out of code block
      await page.keys('ArrowRight');
      await page.keys('Return');
      // Insert a second code block
      await page.click(`[aria-label="${messages.codeblock.defaultMessage}"]`);

      // Make sure the second code block doesn't have a language set.
      await page.waitForSelector(codeBlockSelectors.languageSelectInput);
      const secondCodeblockInitialLanguage = await page.getText(
        floatingToolbarLanguageSelector,
      );
      expect(secondCodeblockInitialLanguage.trim()).toEqual('Select language');
      // Set a language on the second code block
      await page.type(codeBlockSelectors.languageSelectInput, ['Arduino']);
      await page.keys('Return');

      // Check that the language on the first code block is still the same
      await page.click('code[data-language="javascript"]');
      await page.waitForSelector(floatingToolbarLanguageSelector);
      const firstCodeBlock = await page.getText(
        floatingToolbarLanguageSelector,
      );
      expect(firstCodeBlock.trim()).toEqual('JavaScript');

      // Check that the language on the second code block is still the same
      // not sure if this is working 100%
      await page.click('code[data-language="arduino"]');
      await page.waitForSelector(floatingToolbarLanguageSelector);
      const secondCodeBlock = await page.getText(
        floatingToolbarLanguageSelector,
      );
      expect(secondCodeBlock.trim()).toEqual('Arduino');
    },
  );

  BrowserTestCase(
    `code-block: code block content is copied to clipboard after clicking on the copy button for ${editor}`,
    { skip: ['safari', 'firefox', 'edge'] },
    async (client: any, testName: string) => {
      const page = await goToEditorTestingExample(client);

      await mountEditor(page, {
        appearance: editor as EditorAppearance,
        codeBlock: { allowCopyToClipboard: true },
      });

      // Insert code block
      await page.click(`[aria-label="${messages.codeblock.defaultMessage}"]`);

      //Type
      page.pause(100);
      await page.waitForSelector(codeBlockSelectors.content);
      await page.type(codeBlockSelectors.content, 'Some code here');

      // Click on Copy button
      await page.waitForSelector(codeBlockSelectors.copyToClipboardButton);
      await page.click(codeBlockSelectors.copyToClipboardButton);

      // Navigate outside the code block and paste
      await page.keys(['ArrowDown']);

      await page.paste();

      const doc = await page.$eval(editable, getDocFromElement);
      expect(doc).toMatchCustomDocSnapshot(testName);
    },
  );

  // This bug isn't fixed. See ticket for details.
  // https://product-fabric.atlassian.net/browse/ED-7545
  /*
  BrowserTestCase(
    'code-block: code block selected language correctly changes when moving selection directly from one code block to another where one blocks selected is undefined',
    { skip: ['edge', 'safari'] },
    async (client: any, testName: string) => {
      const page = new Page(client);
      await page.goto(editor.path);
      await page.waitForSelector(editor.placeholder);
      await page.click(editor.placeholder);
      // Insert code block
      await insertBlockMenuItem(page, messages.codeblock.defaultMessage);
      await page.waitForSelector(codeBlockSelectors.languageSelectInput);
      // Move out of code block
      await page.keys(['ArrowDown']);
      // Insert a second code block
      await insertBlockMenuItem(page, messages.codeblock.defaultMessage);
      // Make sure the second code block doesn't have a language set.
      await page.waitForSelector(codeBlockSelectors.languageSelectInput);
      const secondCodeblockInitialLanguage = await page.getText(
        floatingToolbarLanguageSelector,
      );
      expect(secondCodeblockInitialLanguage.trim()).toEqual('Select language');
      // Set a language on the second code block
      await page.type(codeBlockSelectors.languageSelectInput, ['C'])
      await page.keys(['Return']);

      // Check that the language on the first code block is still the same
      await page.click('code');
      await page.waitForSelector(floatingToolbarLanguageSelector);
      const firstBlockLanguage = await page.getText(
        floatingToolbarLanguageSelector,
      );
      expect(firstBlockLanguage.trim()).toEqual('Select language');
    },
  );
  */
});
