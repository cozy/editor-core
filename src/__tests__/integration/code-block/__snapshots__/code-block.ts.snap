// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code-block: code block content is copied to clipboard after clicking on the copy button for comment 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "language": null,
        "uniqueId": null,
      },
      "content": Array [
        Object {
          "text": "Some code here",
          "type": "text",
        },
      ],
      "type": "codeBlock",
    },
    Object {
      "content": Array [
        Object {
          "text": "Some code here",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`code-block: code block content is copied to clipboard after clicking on the copy button for full-page 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "language": null,
        "uniqueId": null,
      },
      "content": Array [
        Object {
          "text": "Some code here",
          "type": "text",
        },
      ],
      "type": "codeBlock",
    },
    Object {
      "content": Array [
        Object {
          "text": "Some code here",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`code-block: produces correct ADF after language change for comment 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "language": "javascript",
        "uniqueId": null,
      },
      "type": "codeBlock",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`code-block: produces correct ADF after language change for full-page 1`] = `
Object {
  "content": Array [
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "language": "javascript",
        "uniqueId": null,
      },
      "type": "codeBlock",
    },
  ],
  "type": "doc",
}
`;
